jlib/new.go

```diff
@@ -259,23 +319,32 @@ func ObjectsToDocument(input interface{}) (interface{}, error) {
 		return nil, errors.New("$objectsToDocument input must be an array of objects")
 	}
 
-	output := make(map[string]interface{}) // Initialize the output map
-	// Iterate through each item in the input
+	output := make(map[string]interface{})
 	for _, itemToInterface := range trueInput {
 		item, ok := itemToInterface.(map[string]interface{})
 		if !ok {
-			return nil, errors.New("$objectsToDocument input must be an array of objects with Code and Value fields")
+			return nil, errors.New("$objectsToDocument input must be an array of objects with Code and Val/Value fields")
 		}
-		// Call setValue for each item to set the value in the output map
+
 		code, ok := item["Code"].(string)
 		if !ok {
 			continue
 		}
-		value := item["Value"]
-		setValue(output, code, value)
+
+		// Try "Val" first, then fall back to "Value"
+		var value interface{}
+		if val, exists := item["Val"]; exists {
```

Comment:
is this just a backward compatibility thing? I.e some devs use "Val" and others use "Value"?

Reply:
Assume this to be the case. Now outdated as also added nil check